@shared/api/external.ts

```diff
@@ -13,16 +19,16 @@ export const requestAccess = async (): Promise<string> => {
     console.error(e);
   }
 
-  const { publicKey, error } = response;
+  const { publicKey } = response || { publicKey: "" };
 
-  if (error) {
-    throw error;
-  }
-  return publicKey;
+  return { publicKey, error: response?.apiError };
```

Comment:
A similar change happens in each of the functions below:

instead of throwing on error, we're returning the error as a possible key on the payload